# Data Analysis on Heart Disease

## Introduction
Heart diseases are one of the leading causes of death in the world including the United States for many years. With the increasing rate of cases for heart diseases worldwide, it is important to focus at the leading causes of this disease and establish a relationship between these vicious disease. As a base for our analysis we are going to use the heart disease dataset available at Kaggle. 
(https://www.kaggle.com/datasets/johnsmith88/heart-disease-dataset/data)

This is the cleaned version of the data set given in Canvas -> group_project_proposal


The dataset originates from 1988 and evaluates data from four different databases: Cleveland, Hungary, Switzerland and Long Beach V. While the dataset does have 76 distinct attributes, it only utilizes 14 of them. In our group project, we chose to focus on the variables age, resting blood pressure, cholesterol, and  maximum heart rate and compere it to the diagnosis of heart disease. We chose this variables since according to National Center for Chronic Disease Prevention and Health Promotion (NCCDPHP) it is considered one of the leading reason of causing the heart disease. We will be comparing the numerical variables by the categorical variables (2 cases) “0” and “1” which shows the diagnosis of heart diseases. 

From thee following informations, we will be attempting to answer the following predictive question. 

**Predictive question:**
How does the amount of cholesterol, type of heart defect, age, and sex help us predict the diagnosis of heart disease?

We will be conducting a KNN classification in order to answer the predictive question.

In [6]:
### Run this cell before continuing.
install.packages("kknn")
library(kknn)
library(repr)
library(tidyverse)
library(tidymodels)
library(dplyr)
library(ggplot2)
options(repr.matrix.max.rows = 10)

Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done



### Reading the Data

First read the raw data from GitHub that we uploaded.

In [7]:
heart_disease_data <- read_csv("https://raw.githubusercontent.com/yma24ma/dsci_009_43_gp/main/heart.csv")
heart_disease_data

Rows: 1025 Columns: 14
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
dbl (14): age, sex, cp, trestbps, chol, fbs, restecg, thalach, exang, oldpea...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


age sex cp trestbps chol fbs restecg thalach exang oldpeak slope ca thal
1    52  1   0  125      212  0   1       168     0     1.0     2     2  3   
2    53  1   0  140      203  1   0       155     1     3.1     0     0  3   
3    70  1   0  145      174  0   1       125     1     2.6     0     0  3   
4    61  1   0  148      203  0   1       161     0     0.0     2     1  3   
5    62  0   0  138      294  1   1       106     0     1.9     1     3  2   
⋮    ⋮   ⋮   ⋮  ⋮        ⋮    ⋮   ⋮       ⋮       ⋮     ⋮       ⋮     ⋮  ⋮   
1021 59  1   1  140      221  0   1       164     1     0.0     2     0  2   
1022 60  1   0  125      258  0   0       141     1     2.8     1     1  3   
1023 47  1   0  110      275  0   0       118     1     1.0     1     1  2   
1024 50  0   0  110      254  0   0       159     0     0.0     2     0  2   
1025 54  1   0  120      188  0   1       113     0     1.4     1     1  3   
     target
1    0     
2    0     
3    0     
4    0     
5    0     
⋮    ⋮     
1021 1     
1022 0     
1023 0     
1024 1     
1025 0

**Variables**

age: Age

sex: Sex

cp: Chest pain type (4 values)

trestbps: resting blood pressure

chol: serum cholestoral in mg/dl

fbs: fasting blood sugar > 120 mg/dl

restecg: resting electrocardiographic results (values 0,1,2)

thalacg: maximum heart rate achieved

exang: exercise induced angina

oldpeak: ST depression induced by exercise relative to rest

slope: the slope of the peak exercise ST segment

ca: number of major vessels (0-3) colored by flourosopy

thal: 0 = normal; 1 = fixed defect; 2 = reversable defect

target: diagnosis of heart disease

**Select Data**

We will now be using the `select()` function to select the row that we will be using in this analysis and make it into one table.

In [8]:
heart_disease_selected <- select(heart_disease_data, age, chol, target, thalach, trestbps)|>
                        mutate(target=as_factor(target))|>
                        mutate(heart_disease=fct_recode(target, "Yes" = "1", "No " = "0"))
heart_disease_selected

age chol target thalach trestbps heart_disease
1    52  212  0      168     125      No           
2    53  203  0      155     140      No           
3    70  174  0      125     145      No           
4    61  203  0      161     148      No           
5    62  294  0      106     138      No           
⋮    ⋮   ⋮    ⋮      ⋮       ⋮        ⋮            
1021 59  221  1      164     140      Yes          
1022 60  258  0      141     125      No           
1023 47  275  0      118     110      No           
1024 50  254  1      159     110      Yes          
1025 54  188  0      113     120      No

We use the sum function to check if there are any NA values in our data tables

In [9]:
sum(is.na(heart_disease_selected))

[1] 0

**Average of Selected**

In [ ]:
hd_average1 <- heart_disease_selected |>
                map(mean) 
hd_average1

**Plot Sex X Target**

In [ ]:
hd_sex_target_df <- heart_disease_selected |>
                        group_by(sex) |>
                        summarize(average_target = mean(target))
hd_sex_target_df

In [ ]:
hd_age_chol<-select(heart_disease_selected,age,chol,trestbps,heart_disease)|>
                mutate(heart_disease)
hd_age_chol

In [ ]:
hd_age_chol_plot<-ggplot(hd_age_chol,aes(x=age,y=chol,colour=heart_disease))+
                        geom_point(alpha=0.3)+
                        labs(x="Age",y="Cholesterol",colour="Heart Disease")+       
                    ggtitle("Age vs Cholesterol Scatterplot")
hd_age_chol_plot   

In [ ]:
all_temp_plot <- hd_age_chol|> 
   ggplot(aes(x = age, y = chol)) + 
   geom_point() + 
   facet_wrap(facets = vars(factor(heart_disease, levels = c("No","Yes")))) +
   xlab('Age') + 
   ylab('Cholesterol') +
   theme(text = element_text(size=20))

all_temp_plot

In [ ]:
hd_age_thalach<-select(heart_disease_selected,age,thalach,heart_disease)|>
                mutate(heart_disease)
hd_age_thalach

In [ ]:
hd_age_thalach_plot<-ggplot(hd_age_thalach,aes(x=age,y=thalach,colour=heart_disease))+
                        geom_point(alpha=0.3)+
                        labs(x="Age",y="Maximum Heart Rate",colour="Heart Disease")+       
                    ggtitle("Age vs Maximum Heart Rate Scatterplot")
hd_age_thalach_plot   

In [ ]:
all_temp_plot <- hd_age_thalach |> 
   ggplot(aes(x = age, y = thalach)) + 
   geom_point() + 
   facet_wrap(facets = vars(factor(heart_disease, levels = c("No","Yes")))) +
   xlab('Age') + 
   ylab('Maximum heat rate') +
   theme(text = element_text(size=20))

all_temp_plot

In [ ]:
hd_age_trestbps<-select(heart_disease_selected,age,trestbps,heart_disease)|>
                mutate(heart_disease)
hd_age_trestbps

In [ ]:
hd_age_trestbps_plot<-ggplot(hd_age_trestbps,aes(x=age,y=trestbps,colour=heart_disease))+
                        geom_point(alpha=0.3)+
                        labs(x="Age",y="Resting Blood Pressure(mm/Hg)",colour="Heart Disease")+       
                    ggtitle("Age vs Resting Blood Pressure Scatterplot")
hd_age_trestbps_plot   

In [ ]:
all_temp_plot <- hd_age_trestbps |> 
   ggplot(aes(x = age, y = trestbps)) + 
   geom_point() + 
   facet_wrap(facets = vars(factor(heart_disease, levels = c("No","Yes")))) +
   xlab('Age') + 
   ylab('Resting Blood Pressure') +
   theme(text = element_text(size=20))

all_temp_plot

## Methods
For our Heart Disease data set, we are going to use the method of K-nearest neighbors classification. Essentially, we are going to use predictor variables chol (amount of cholesterol), thal (type of heart defect), age and sex to predict the diagnosis class of heart disease, which can be categorized into 0 (no heart disease) or 1 (heart disease). Therefore, the column names we will incorporate are chol, thal, age, sex and target. We chose to only use four predictor variables because we think that there are more than two factors that contribute to the diagnosis of heart disease. Since there are multivariables, we can avoid a 4D graph by using the facet_grid function to create a plot that has multiple subplots arranged in a grid.

## Expected outcomes and significance


What do you expect to find?

We expect to identify the most relevant features that contribute to the presence or absence of heart disease.We also look for patterns and correlations within the data.

What impact could such findings have?

Understanding the factors that contribute to heart disease can inform public health initiatives.Furthermore, discoveries from this dataset can enhance healthcare by improving diagnostic tools and predictive models for heart disease, potentially leading to early detection and treatment.

What future questions could this lead to?

Are there additional attributes that should be considered, or are there redundant variables that can be eliminated to improve model performance?



In [ ]:
set.seed(9999) 
heart_disease_recipe <- recipe(heart_disease ~ age + chol + thalach + trestbps,data = heart_disease_selected) |>
                       step_scale(all_predictors()) |>
                       step_center(all_predictors())
                        
heart_disease_scaled <- heart_disease_recipe |>  
                           prep() |> 
                           bake(heart_disease_selected)
heart_disease_scaled

In [ ]:
options(repr.plot.width = 5, repr.plot.height = 5)


heart_disease_chol_plot<-ggplot(heart_disease_scaled,aes(x=age,y=chol,color=heart_disease))+
geom_point(alpha=0.3)+
labs(x="age",
               y="Cholesterol(mg/dl)",
               colour="Heart Disease")


heart_disease_chol_plot

heart_disease_maxheart_plot<-ggplot(heart_disease_scaled,aes(x=age,y=thalach,color=heart_disease))+
geom_point(alpha=0.3)+
labs(x="age",
               y="Max Heart Rate (bpm)",
               colour="Heart Disease")


heart_disease_maxheart_plot

heart_disease_rbp_plot<-ggplot(heart_disease_scaled,aes(x=age,y=trestbps,color=heart_disease))+
geom_point(alpha=0.3)+
labs(x="age",
               y="Resting Blood Pressure(mm/Hg)",
               colour="Heart Disease")


heart_disease_rbp_plot